doc/nrf/protocols/matter/getting_started/adding_clusters.rst

```diff
@@ -159,6 +159,15 @@ To edit clusters using the ZAP tool, complete the following steps:
 
 At this point, new clusters have been added to the Matter device.
 
+.. note::
+   On the first run the ZAP tool creates :file:`.zap` directory to store cached information for the following runs.
+   The default directory location is User's Home directory and it can be overriden by adding ``--stateDirectory`` and location path to the invoked ZAP commands.
+
+   Introducing significant changes to the ZAP tool configuration, like updating the tool version or changing the used zcl templates, may result in unexpected issues with the application that uses information previously cached in the :file:`.zap` directory.
+   The application behavior is undefined and it depends on the difference between the new configuration and old cached data.
+   For example, it could be problems with displaying specific information in the UI, generating new configuration or even application crashes. 
```

Comment:
```suggestion
   For example, it could result in problems with displaying specific information in the UI, generating new configuration, or even application crashes. 
```